### ISSUE TITLE
How to add step chain helpers?

### ISSUE THREAD

**Issue by lorensr** [xolvio/meteor-cucumber#80]
Would be nice to see this in docs. For example `.waitForVisible(X).click(X)` seems to happen a lot

https://github.com/meteor-velocity/velocity-examples/blob/master/leaderboard-cucumber/tests/cucumber/features/step_definitions/steps.js#L21-L33

and I'd like to make a `.waitAndClick` function, but don't know the right way to define it so that it fits with the webdriver/chai function chain.

**Comment by djhi** [comment]
You must implement  custom commands for that:
http://webdriver.io/guide/usage/customcommands.html

I struggled a bit to use them with this package. I'll try to make a guide
about it.

@*samhatoum*: a wiki might be nice ;)

**Comment by Sanjo** [comment]
@djhi Wiki: https://github.com/xolvio/meteor-cucumber/wiki. I think everyone can edit content there.

**Comment by djhi** [comment]
Ok i'll add something about this when i get the chance

**Comment by lorensr** [comment]
Great, thanks!

**Comment by samhatoum** [comment]
@lorensr did the above work for you?

**Comment by lorensr** [comment]
Waiting on Guildas' wiki guide

**Comment by djhi** [comment]
Sorry about the delay, lots of work...

I have troubles with my commands. They don't work anymore.. I get `Problem: Can't find variable: Meteor` now, with all commands calling Meteor

**Comment by djhi** [comment]
For reference, here's how I add my custom commands:

In my `tests/cucumber/features/support/hooks.coffee` file:
```CoffeeScript
@Before ->
    callback = arguments[arguments.length - 1]
    console.log "Adding custom Commands"
    commands = require './commands'
    commands.addCommands global.browser
    global.browser.call callback
```

In `tests/cucumber/features/support/commands/index.coffee`:
```CoffeeScript
fs = require 'fs'

exports = module.exports = {}

exports.addCommands = (browser) ->
    # Loop over all files in this folder
    fs
        .readdirSync(__dirname + '/')
        .forEach (file) ->
            if file.match(/.+\.coffee/g) != null and file != 'index.coffee'
                # require the command
                name = file.replace('.coffee', '')
                command = require('./' + file)

                # add it to the browser
                command.addCommand browser
            return
```

And this is one of the commands:
```CoffeeScript
exports = module.exports = {}

exports.addCommand = (browser) ->
    browser.addCommand "logout", ->
        callback = arguments[arguments.length - 1]
        @executeAsync (callback) ->
            Meteor.logout (err) -> callback()
            return
        ,
            callback

    return
```

**Comment by samhatoum** [comment]
@djhi what do you mean by this:

>  I get Problem: Can't find variable: Meteor now, with all commands calling Meteor

Let me know if you need help by filing another issue

**Comment by samhatoum** [comment]
@lorensr can I close this now?

**Comment by lorensr** [comment]
@djhi Does your `@Before` still work? When I try, I'm getting: 

![image](https://cloud.githubusercontent.com/assets/251288/7827496/1d6d6584-03dd-11e5-8f93-9f624f013fb1.png)

@samhatoum I'm guessing the problem is due to his upgrading to a version of cucumber after you removed the Meteor context from the environment in which test files are run.

**Comment by samhatoum** [comment]
Hey @djhi did you see this? https://github.com/xolvio/meteor-cucumber/issues/114 Might help with your "Can't find Meteor" issue.

Hey @lorensr `this.Before` should work, just be sure to include it in a `module.exports = function () {...` and I think the Meteor instance @djhi is referring to is the one that's inside the executeAsync so should be in the browser

**Comment by lorensr** [comment]
Thanks, that worked. And here is an eg of composing webdriver cmds:

```coffeescript
exports = module.exports = {}

exports.addCommand = (browser) ->
  browser.addCommand "waitAndClick", (selector) ->
    callback = arguments[arguments.length - 1]
    @waitForVisible selector
      .click selector
      .call callback
```

**Comment by samhatoum** [comment]
mm, I like egs :) Thanks!

**Comment by djhi** [comment]
@samhatoum 
>  I think the Meteor instance @djhi is referring to is the one that's inside the executeAsync so should be in the browser

Right.

> Hey @djhi did you see this? #114 Might help with your "Can't find Meteor" issue.

Going to check that right now :)

**Comment by djhi** [comment]
No, I'm no trying to run code on a statically served page. 

This is the feature step:
```gherkin
    Given I am a new Visitor
```

This is the code for it:
```coffeescript
module.exports = ->
    url = require('url')
    expect = require('chai').expect
    should = require('chai').should()

    @Given /^I am a new Visitor$/, (callback) ->
        root = url.resolve process.env.ROOT_URL, "/"
        @browser
            .url root
            .logout()
            .call callback
```

The logout command is the one I posted above. I get the can't find Meteor error in the executeAsync

**Comment by samhatoum** [comment]
Things to try

- Manually go to the mirror and check the console
- tame a screenshot from the step before execute async
- add a pause to see if that changes anything
- return a debug message from the async call to the done function

**Comment by djhi** [comment]
Thx for your time sam. I checked the mirror to see if any error happened. None was reported and Meteor is available in the global scope.

**Comment by samhatoum** [comment]
If the mirror works normally, the we need to see what WebdriverIO is seeing. I would try to do a screenshot next, and also from within the async, return some diagnostic values using the async callback, and inspect the result in the executeAsync result params. Also check the executeAsync error object.

**Comment by djhi** [comment]
For info, same code works fine using `CHIMP_OPTIONS='--browser=chrome'`

**Comment by samhatoum** [comment]
does the same thing work if you do:

`SELENIUM_BROWSER=chrome` or only with `CHIMP_OPTIONS` ?

**Comment by djhi** [comment]
Only with CHIMP_OPTIONS

**Comment by samhatoum** [comment]
ok, let me know if you can isolate this error, or if I can help do that
